exchange/utils.go

```diff
@@ -227,21 +234,32 @@ func getAuctionBidderRequests(req AuctionRequest,
 		}
 	}
 
-	reqExt, err := getExtJson(req.BidRequest, requestExt)
-	if err != nil {
-		return nil, []error{err}
-	}
-
 	var errs []error
 	for bidder, imps := range impsByBidder {
 		coreBidder := resolveBidder(bidder, aliases)
 
 		reqCopy := *req.BidRequest
 		reqCopy.Imp = imps
-		reqCopy.Ext = reqExt
 
 		prepareSource(&reqCopy, bidder, sChainsByBidder)
 
+		if len(bidderExt) != 0 {
+			if bidderParams, ok := bidderExt[bidder]; ok {
+				if requestExt == nil {
+					requestExt = &openrtb_ext.ExtRequest{}
+				}
+				requestExt.Prebid.BidderParams = bidderParams
+			} else if requestExt != nil {
+				requestExt.Prebid.BidderParams = nil
+			}
+		}
```

Comment:
It's difficult for me to understand what this block of code is doing. Consider extracting to a method and use it's name to convey purpose / meaning.

Reply:
Currently I am merging request level bidder params at both new (imp[].ext.prebid.bidder) level & old (imp[].ext) level.
Let me know if I should remove support for old level from pull request.

Reply:
It must be confusing as I am reading this block of code as doing something completely different. It is modifying the `request.ext.prebid.bidderparams` of the bidder specific copy of a request to only contain the bidder parameters of the bidder in questions, hiding info about all other bidders from this particular bidder. Some comments or removal to its own function may make this all clearer.

Reply:
Added comment in code block to explain the logic.